rsciio/topspin/_api.py

```diff
@@ -0,0 +1,432 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2025 The HyperSpy developers
+#
+# This file is part of RosettaSciIO.
+#
+# RosettaSciIO is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# RosettaSciIO is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RosettaSciIO. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+
+import datetime
+import os
+import sys
+import xml.etree.cElementTree as ET
+
+import h5py
+import numpy as np
+from tqdm import tqdm
+
+from rsciio._docstrings import FILENAME_DOC, RETURNS_DOC
+
+
+def file_reader(filename, subset=None, dryrun=False, show_progressbar=True):
+    """
+    Read .app5 file format used by NanoMegas's Topspin software.
+
+    .app5 files use the hdf5 file format, with metadata stored as
+    binarized XML-style text strings.
+
+
+    Parameters
+    ----------
+    %s
+    subset: str or None
```

Comment:
> I am not sure what is meant with "h5py-style address". Maybe some an example inline here or in the example section?

 "h5py Group name" is a better name perhaps? Its the location inside the hdf5 file where the data is stored. I will clarify.

Reply:
looks like `dataset_path` is the equivalent for the NX reader. Not an exact one-to-one, but close enough to justify mimicking the variable name. 

Fixed.